packages/authentication/src/authentication.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import http = require('http');
+
+/**
+ * Authentication class to define the lifecycle of a passport strategy.
+ * Instance is created by passing a passport strategy in the constructor
+ */
+export class Authentication {
+  private strategyCtor: {};
+
+  constructor(strategy) {
+    this.strategyCtor = strategy;
+  }
+
+  /**
+   * The function to invoke the contained passport strategy.
+   *     1. Create an instance of the strategy
+   *     2. add success and failure state handlers
+   *     3. authenticate using the strategy
+   * @param req {http.ServerRequest} The incoming request.
+   * @param res {http.ServerResponse} The response object.
+   * @param next {Function} callback for authentication.
+   */
+  authenticate(req, res, next) {
```

Comment:
I would expect when using TypeScript to have parameter types and return type, in this entire code I don't see any type which lost the sense of using TypeScript